Контрольная по ударениям

Maksonchik88 opened the pull request:
https://informatics.msk.ru/mod/statements/view.php?id=4535&chapterid=3768#1 

*Открыл файл
*Инициализировал словарь, где ключ - слово в без ударений, а значение - список слов с обозначенными ударениями *Инициализировал список слов, которые нужно проверить
 *Инициализировал словарь, в котором ключ - проверяемое слово, значение - количество ударений в слове 
*Циклом проверил, сколько ударений с проверяемом слове. Если более одного или ни одного, то это ошибка

*Не разобрался, как проверить слово в значениях первого словаря....
*Не убрал отладочные принты Lol

Maksonchik88 committed:
Контрольная по ударениям

https://informatics.msk.ru/mod/statements/view.php?id=4535&chapterid=3768#1
*Открыл файл
*Инициализировал словарь, где ключ - слово в без ударений, а значение - список слов с обозначенными ударениями
*Инициализировал список слов, которые нужно проверить
*Инициализировал словарь, в котором ключ - проверяемое слово, значение - количество ударений в слове
*Циклом проверил, сколько ударений с проверяемом слове. Если более одного или ни одного, то это ошибка

*Не разобрался, как проверить слово в значениях первого словаря....

iliadmitriev commented on dict/accent_test.py:
```diff
@@ -0,0 +1,32 @@
+
+data_completed = {}
+for word in completed_exercise:
+    letters_counter = 0
+    for ch in word:
+        if ch.isupper():
+            letters_counter += 1
```
это предложение (не обязательно)
можно вот эти 4 строки заменить на генераторное выражение.
хорошая практика будет.

Maksonchik88 replied:
а как?

iliadmitriev commented on dict/accent_test.py:
```diff
@@ -0,0 +1,32 @@
+file = open("dict/text.txt")
+word_count = int(file.readline())
+data = {}
```
более осознанные имена переменным надо дать, 
data это что то абстрактное.

iliadmitriev commented on dict/accent_test.py:
```diff
@@ -0,0 +1,32 @@
+mistakes = 0
+for word in completed_exercise:
+    word_lower = word.lower()
+    if data_completed[word] > 1 or data_completed[word] == 0:
+        mistakes += 1
```
Хорошая идея проверить сначала, не стоит ли в слове ударение или стоит больше 2-х ударений и считать автоматически это ошибкой.
Но вот нет альтернативного сценария, когда в слове единственное ударение, то нужно по результатам проверки его наличия (или отсутствия) в заранее подготовленном словаре, принимать решение о том является ли это ошибкой или нет.
А то так получается что весь предыдущий код не нужен, иначе зачем ты делал словарик и не используешь его.
Судя по комментам, я так понял задачу просто не дописал.

Maksonchik88 replied:
да, не дописал

iliadmitriev replied:
Давай дописывай, пару строк осталось до решения 😉

iliadmitriev reviewed (changes requested):
Надо доделать

Maksonchik88 committed:
*В первом словаре в занчении создал множества
*Добавил в конечный цикл условие,  при котором проверяется так же наличие проверяемого слова в значениях словаря

iliadmitriev commented on dict/accent_test.py:
```diff
@@ -0,0 +1,32 @@
+for word in completed_exercise:
+    lower_word = word.lower()
+    if lower_word in words_from_dictionary and word not in words_from_dictionary[lower_word] and data_completed[
+        word] > 1 or data_completed[word] == 0:
```
Задача решена не верно.

iliadmitriev reviewed (changes requested):
Задача решена не верно.
Для тестовых данных:
```
4
cAnnot
cannOt
fOund
pAge
thE pagE cAnnot be found
```
Программа должна выдать 3: в 2-х словах `be` `found` ударение не стоит, а в слове `pagE` ударение не верное

Программа выдает ответ 2

Maksonchik88 committed:
*Видоизменил алгоритм решения в последнем цикле, согласно рекомендациям

iliadmitriev reviewed (approved):
отличная работа!
